src/averageAges.js

```diff
@@ -55,7 +73,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children;
+
+  if (onlyWithSon) {
+    children = people.filter(el => el.mother && el.sex === 'm');
+  } else {
+    children = people.filter(el => el.mother);
+  }
+
+  const agesDiff = [];
+
+  for (const child of children) {
```

Comment:
The note about loops also applies to all 3 tasks. Referring to Vitalii Arseniuk's answer